### PR TITLE
add relative path support for file

### PR DESCRIPTION
**Pull request by yupbank**
for convenience of relative file paths

### CODE REVIEW

**Comment by tmylk** [comment]
Thanks for the PR! 
Can we make it optional(on by default)? 

The tests fail as they don't expect an expansion yet. Need tests for both.

**Comment by yupbank** [comment]
I think on by default would be nice, since it is not conflicting with other schema :)

And i have updated the abspath which is not needed for non expanding case

**Comment by tmylk** [comment]
@yupbank Are you planning to add tests for this new functionality? And a note in Changelog?

**Comment by yupbank** [comment]
hey, i have added a few tests and modified the change log :)

File: smart_open/smart_open_lib.py [hunk 1]
```diff
@@ -241,13 +248,23 @@ def __init__(self, uri, default_scheme="file"):
...
             self.uri_path = parsed_uri.netloc + parsed_uri.path
+            self.uri_path = expand_full_path(self.uri_path)
 
             if not self.uri_path:
                 raise RuntimeError("invalid file URI: %s" % uri)
         else:
             raise NotImplementedError("unknown URI scheme %r in %r" % (self.scheme, uri))
 
 
+def expand_full_path(path):
```
Comment (piskvorky) [inline smart_open/smart_open_lib.py]: What is this for? Why create an alias to a built-in function?

**Comment by tmylk** [comment]
@yupbank  Thanks for the PR!

**Comment by piskvorky** [comment]
I don't see any changes to CHANGELOG in this PR.

**Pull request merged**